rewrite/to_ir.rs

```diff
@@ -119,109 +41,145 @@ impl<'i> ParseTreeVisitor<'i, CypherParserContextType> for FilterVisitor {
 }
 
 impl<'i> CypherVisitor<'i> for FilterVisitor {
-    /// This function visits a match clause.  It extracts the graph inside, complete with any
-    /// node attributes, and stores that information in a struct_filter.  It then extracts any information
-    /// in the where clause, which pertains to the whole graph, and stores that in an attribute_filter.
-    fn visit_oC_Match(&mut self, ctx: &OC_MatchContext<'i>) {
+    // we do not want to visit returns in this case, ignore that part of the tree
+    // TODO: Apply the visitor directly to the MATCH body instead
+    fn visit_oC_Return(&mut self, _ctx: &OC_ReturnContext<'i>) {}
+
+    fn visit_oC_PropertyOrLabelsExpression(
+        &mut self,
+        ctx: &OC_PropertyOrLabelsExpressionContext<'i>,
+    ) {
+        log::info!("Generating IrReturn");
+        let atom = ctx.oC_Atom().unwrap();
+        let entity: String;
+        if let Some(func) = atom.oC_FunctionInvocation() {
+            entity = func.get_text();
+            // TODO: We can make this UDF more precise
```

Comment:
What do you mean by more precise?

Reply:
Right now we get the UDF as a whole string, e.g., `"avg(a.response_size)"`. We may want to split this into function call ("avg", the actual function) and args ("a.response_size", which is a node property).